armi/cli/tests/test_runEntryPoint.py

```diff
@@ -32,38 +33,71 @@
 from armi.cli.run import RunEntryPoint
 from armi.cli.runSuite import RunSuiteCommand
 from armi.physics.neutronics.diffIsotxs import CompareIsotxsLibraries
-from armi.tests import mockRunLogs, TEST_ROOT
+from armi.tests import mockRunLogs, TEST_ROOT, ARMI_RUN_PATH
 from armi.utils.directoryChangers import TemporaryDirectoryChanger
+from armi.utils.dynamicImporter import getEntireFamilyTree
+
+
+class TestInitializationEntryPoints(unittest.TestCase):
+    def test_entryPointInitialization(self):
+        """Tests the initialization of all subclasses of `EntryPoint`."""
+        entryPoints = getEntireFamilyTree(EntryPoint)
+
+        # Note that this is a hard coded number that should be incremented
+        # if a new ARMI framework entry point is added. This is a bit hacky,
+        # but will help demonstrate that entry point classes can be initialized
+        # and the `addOptions` and public API is tested.
+        self.assertEqual(len(entryPoints), 18)
+
+        for e in entryPoints:
+            entryPoint = e()
+            entryPoint.addOptions()
+            settingsArg = None
+            if entryPoint.settingsArgument is not None:
+                for a in entryPoint.parser._actions:
+                    if "settings_file" in a.dest:
+                        settingsArg = a
+                        break
+                self.assertIsNotNone(
+                    settingsArg,
+                    msg=(
+                        f"A settings file argument was expected for {entryPoint}, "
+                        f"but does not exist. This is a error in the EntryPoint "
+                        f"implementation."
+                    ),
+                )
 
 
 class TestCheckInputEntryPoint(unittest.TestCase):
     def test_checkInputEntryPointBasics(self):
         ci = CheckInputEntryPoint()
         ci.addOptions()
-        ci.parse_args(["/path/to/fake.yaml"])
```

Comment:
What's with this change?

I'm not complaining or anything.  I just don't understand.

Reply:
I'm fairly certain it failed with the fake path, let me go double check that

Reply:
Nope I'm wrong

That must have been a mistake! I'll push up a change. I feel like we should only use legit files when we need legit files

Reply:
Your call. I was seriously just asking.

Reply:
It was unnecessary. Leftover from troubleshooting most likely, already switched!